### PR TITLE
Bug/cleanup after tests

### LINKED ISSUE #331: Minimesos doesn't cleanup after exceptions during tests
It seems that any failed test / or failed containers leaves the cluster running.
 There should be a clean-up routine that is executed when an unfixable cluster state is reached.
(eg can't pull one of the images)

### PR DESCRIPTION
**Pull request by alex-glv**
fixes #331 
- Replacing ```@After``` calls with cluster.destroy();
- helloworld container gets cluster prefix in the name, so cluster.destroy() removes it, too.

### CODE REVIEW

**Comment by frankscholten** [comment]
LGTM

File: minimesos/src/test/java/com/containersol/minimesos/DefaultMinimumClusterTest.java [hunk 1]
```diff
@@ -38,8 +38,7 @@
 
     @After
     public void after() {
-        DockerContainersUtil util = new DockerContainersUtil(dockerClient);
-        util.getContainers(false).filterByName(HelloWorldContainer.CONTAINER_NAME_PATTERN).kill().remove();
+        cluster.destroy();
```
Comment (sadovnikov) [inline minimesos/src/test/java/com/containersol/minimesos/DefaultMinimumClusterTest.java]: * Are you sure this is enough? Containers, which are started by scheduler, are not part of the cluster. Therefore they are not stopped. If I recall correctly, the scheduler gets killed by `destroy()`, but schedulers remain running
* CLUSTER is annotated by `@ClassRule`, therefore `stop()` will be call through `MesosCluster.after()`

**Comment by sadovnikov** [comment]
@alex-glv tests on this branch fail on my machine and on Jenkins https://ci.infra.container-solutions.com/job/minimesos_TestsAndReports_branch/1/

File: minimesos/src/test/java/com/containersol/minimesos/DefaultMinimumClusterTest.java (hunk 1 above)
Comment (alex-glv) [inline minimesos/src/test/java/com/containersol/minimesos/DefaultMinimumClusterTest.java]: I don't think we have any control of the containers that are started by schedulers (ie we can't give them the name prefixed with minimesos-<CLUSTERID>) and that's the only way for us to identify if the containers belongs to a cluster.

So, for this case it's suffice to call cluster.destroy(), and not a single test starts schedulers that start containers, so I don't think it's a problem

**Comment by alex-glv** [comment]
@sadovnikov Have the tests been running ok since latest commits? Can we close this one?

**Comment by sadovnikov** [comment]
@alex-glv one `minimesos-test-1536702375-3740104270` container left running after completion of the build... will investigate further...

**Comment by frankscholten** [comment]
@alex-glv This one does not compile because `log4j.Logger` cannot be found

**Comment by alex-glv** [comment]
@frankscholten can you give more details, please? There are now log4j related in this branch at all, that's strange

**Comment by frankscholten** [comment]
@alex-glv Never mind problem with my setup

**Comment by sadovnikov** [comment]
Found the problem, which was seen in `RunTaskTest` - container has exited during the test, but was not removed after the test completion. Therefore I merged two different `MesosCluster` methods with the same meaning: `stop()` and `destroy()`

**Comment by sadovnikov** [comment]
Approved

**Comment by sadovnikov** [comment]
Please, **DO NOT MERGE** until #371 is in

**Comment by frankscholten** [comment]
LGTM

**Pull request merged**